src/environmentd/tests/server.rs

```diff
@@ -2823,9 +2823,13 @@ fn test_cancel_read_then_write() {
             });
             std::thread::sleep(Duration::from_millis(100));
             let handle2 = thread::spawn(move || {
-                client2
+                let err = client2
                 .batch_execute("insert into foo values ('blah', 1);")
-                .unwrap();
+                .unwrap_err();
+                assert_contains!(
+                    err.to_string(),
+                    "canceling statement"
+                );
```

Comment:
This is the only test that started failing as a result of the change here: https://github.com/MaterializeInc/materialize/pull/24826/files#r1481988333 (switching from sending `Ok(ExecuteResponse::Canceled)` to `Err(AdapterError::Canceled)`).

As you can see the big difference is that the client used to receive an `Ok` response and now they receive an `Err` response. IMO `Err` is the correct semantics here. If you read this test you'll see that previously it was expecting this `INSERT` to both fail and send an `Ok` response, which doesn't make a lot of sense. Now it fails and sends an `Err` response which makes much more sense to me.